broker-core/src/main/java/org/apache/qpid/server/transport/NonBlockingConnectionTLSDelegate.java

```diff
@@ -310,6 +337,11 @@ private int wrapBufferArray(Collection<QpidByteBuffer> buffers) throws SSLExcept
         }
         while(encrypted && _sslEngine.getHandshakeStatus() != SSLEngineResult.HandshakeStatus.NEED_UNWRAP);
 
+        if (enableDiagnosisOfSslEngineLooping)
```

Comment:
Daniil,
Thanks for adding  a reset for a loop counter...
Though, I am wondering whether the counter reset should be made conditional... 
As far as I remember issues reported as part of QPID-8545 and QPID-8526, the connection was looping in `NonBlockingConnection`, `NonBlockingConnectionTLSDelegate`, etc
Thus, it seems that counter should be reset only if data was actually written into the output buffer... Otherwise, you might not detect an engine  issue when encryption loop ends without writing any data...
What do you think?

Reply:
Hi Alex,
I agree, connection looping was happening inside of both `NonBlockingConnection` and `NonBlockingConnectionTLSDelegate`. It also happened either when initiating SSL connection or when disconnecting. So checking whether data was written into the output buffer or not seems to be right solution.